dev/CMakeLists.txt

```diff
@@ -240,6 +240,37 @@ if (RESTINIO_GCC_CODE_COVERAGE)
 #     cmake --build _build --target RESTINIO_CODE_COVERAGE_all
 endif()
 
+option(RESTINIO_CLANG_TIDY "Scan build with cllang-tidy" OFF)
+if (RESTINIO_CLANG_TIDY)
+    message("========================================")
+    message(STATUS "Enabling clang-tidy")
+
+    include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/find_program_required.cmake)
+    find_program_required(
+        NAME clang-tidy
+        SEARCH_NAME ${TOOLING_CLANG_TIDY_PATH}
+        PATH clang_tidy_exe )
+
+    if (NOT TOOLING_CLANG_TIDY_PARAMS)
+        # TODO: consider appropriate defaults.
+        string(CONCAT TOOLING_CLANG_TIDY_PARAMS
+            "--header-filter='${CMAKE_SOURCE_DIR}/restinio/.*';"
+            "--warnings-as-errors=*,-cppcoreguidelines-avoid-magic-numbers,-readability-magic-numbers,-altera-struct-pack-align,-cppcoreguidelines-non-private-member-variables-in-classes,-misc-non-private-member-variables-in-classes,-abseil-string-find-str-contains;"
```

Comment:
Can the long lines be formatted more readable? Something like that:

```
"--warnings-as-errors=*,"
  "-cppcoreguidelines-avoid-magic-numbers,"
  "-readability-magic-numbers,"
  "-altera-struct-pack-align,"
```
I think this will made list of options obvious.

Reply:
I'm afraid it would take lots of lines.

What about adding `dev/.clang-tidy-flags` file that will contain these flags and clang-tidy snipped will get flags from the file?

Reply:
> I'm afraid it would take lots of lines.

It's not a problem as for me. It will allow to comment out some unneeded options easily.

> What about adding dev/.clang-tidy-flags file that will contain these flags and clang-tidy snipped will get flags from the file?

Will this file be under the version control?
Does it allow to comment individual options inside and uncomment them later?

Reply:
@eao197 
I make it a cmake-file, so comments are possible.